### PR TITLE
[jrubyscripting] document how to use Ruby in a transformation

### PR DESCRIPTION
**Pull request by ccutrer**
using the core ScriptTransformationService. See https://github.com/openhab/openhab-addons/pull/13255 for prior discussion on this.

### CODE REVIEW

**Comment by jlaur** [comment]
@ccutrer - well done! Just one question: Does this depend on openhab/openhab-core#2872?

**Comment by ccutrer** [comment]
No. That's only if you're using a transform as a channel profile. No changes are needed to use it for item state descriptions.

**Comment by jlaur** [comment]
@boc-tothefuture - I'll let you have a look (if you want) since I'm not into Ruby. :)

File: bundles/org.openhab.automation.jrubyscripting/README.md
Comment (jlaur) [inline bundles/org.openhab.automation.jrubyscripting/README.md:91]: Would this look better?
```suggestion
#### compass.script
```

File: bundles/org.openhab.automation.jrubyscripting/README.md
Comment (jlaur) [inline bundles/org.openhab.automation.jrubyscripting/README.md:105]: With this also:
```suggestion
#### weather.items
```

File: bundles/org.openhab.automation.jrubyscripting/README.md
Comment (jlaur) [inline bundles/org.openhab.automation.jrubyscripting/README.md:87]: One sentence per line:
```suggestion
**Note**: Due to an [issue](https://github.com/jruby/jruby/issues/5876) in the current version of JRuby, you will need to begin your script with `input ||= nil` (and `a ||= nil` etc. for additional query variables) so that JRuby will recognize the variables as variables--rather than method calls--when it's parsing the script.
Otherwise you will get errors like `(NameError) undefined local variable or method 'input' for main:Object`.
```

File: bundles/org.openhab.automation.jrubyscripting/README.md
Comment (jimtng) [inline bundles/org.openhab.automation.jrubyscripting/README.md:85]:
```suggestion
You can also specify additional variables to be set in the script using a URI-like query syntax: `SCRIPT(rb:mytransform.script?a=1b=c):%s` in order to share a single script with slightly different parameters for different items.
```

**Review by jlaur — APPROVED** [review]
Thanks for pioneering this!

**Comment by jlaur** [comment]
@jimtng - would you like to have a look before merging?

**Comment by jimtng** [comment]
I tried following the instructions in the README by adding the item and CONF/transform/compass.script. However, I encountered this error:

```
21:28:44.023 [ERROR] [.ui.internal.items.ItemUIRegistryImpl] - transformation throws exception [transformation=org.openhab.core.automation.module.script.ScriptTransformationService@4049fb61, value=97]
org.openhab.core.transform.TransformationException: Could not get script for UID 'compass.script'.
        at org.openhab.core.automation.module.script.ScriptTransformationService.transform(ScriptTransformationService.java:108) ~[?:?]
        at org.openhab.core.ui.internal.items.ItemUIRegistryImpl.transform(ItemUIRegistryImpl.java:551) ~[?:?]
        at org.openhab.core.ui.internal.items.ItemUIRegistryImpl.getLabel(ItemUIRegistryImpl.java:459) ~[?:?]
        at org.openhab.core.io.rest.sitemap.internal.PageChangeListener.constructSitemapEventForWidget(PageChangeListener.java:236) ~[?:?]
        at org.openhab.core.io.rest.sitemap.internal.PageChangeListener.constructSitemapEvents(PageChangeListener.java:225) ~[?:?]
        at org.openhab.core.io.rest.sitemap.internal.PageChangeListener.constructSitemapEvents(PageChangeListener.java:214) ~[?:?]
        at org.openhab.core.io.rest.sitemap.internal.PageChangeListener.constructAndSendEvents(PageChangeListener.java:172) ~[?:?]
        at org.openhab.core.io.rest.sitemap.internal.PageChangeListener.stateChanged(PageChangeListener.java:186) ~[?:?]
        at org.openhab.core.items.GenericItem.lambda$1(GenericItem.java:259) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
21:28:44.026 [WARN ] [.rest.core.item.EnrichedItemDTOMapper] - Failed transforming the state '97 °' on item 'Exterior_WindDirection' with pattern 'SCRIPT(rb:compass.script):%s': Could not get script for UID 'compass.script'.
```

**Comment by ccutrer** [comment]
Hmm. I actually had problems while I was testing things that renaming files in transform/ wouldn't get picked up, but changing a line and then saving would. I figured it was just something weird on my system, but maybe there is an underlying bug that needs to be found. Anyway, can you try adding like a newline, then saving, and see if that fixes it? You can also use the /rest/transforms API to ensure OpenHAB has loaded the file (and how it interpreted it).

**Comment by jimtng** [comment]
I restarted openhab and now I cannot reproduce the problem. It seems to work fine now. My apologies.

**Pull request merged**
